src/main/scala/com/fulcrumgenomics/testing/VariantContextSetBuilder.scala

```diff
@@ -0,0 +1,163 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package com.fulcrumgenomics.testing
+
+import java.nio.file.Files
+
+import dagr.commons.CommonsDef._
+import htsjdk.samtools.SAMSequenceDictionary
+import htsjdk.variant.variantcontext._
+import htsjdk.variant.variantcontext.writer.{Options, VariantContextWriterBuilder}
+import htsjdk.variant.vcf.{VCFFileReader, VCFHeader, VCFHeaderLine}
+
+import scala.collection.JavaConversions._
+import scala.collection.mutable.ListBuffer
+
+object VariantContextSetBuilder {
+  def apply(sampleName: String): VariantContextSetBuilder = {
+    new VariantContextSetBuilder(List(sampleName))
+  }
+}
+
+/**
+  * A class to make artificial variant contexts and VCFs just a little bit easier to generate.
+  *
+  * This builder uses the default sequence dictionary from [SamRecordSetBuilder] by default.
+  */
+class VariantContextSetBuilder(sampleNames: Seq[String] = List("Sample")) extends Iterable[VariantContext] {
+
+  if (sampleNames.isEmpty) throw new IllegalArgumentException("At least one samle name must be given")
+
+  private var _header = {
+    val h = new VCFHeader(Set.empty[VCFHeaderLine], sampleNames.toList)
+    h.setSequenceDictionary(new SamRecordSetBuilder().header.getSequenceDictionary)
+    h
+  }
+
+  private var variants = new ListBuffer[VariantContext]()
+
+  def header: VCFHeader = this._header
+
+  /** Sets the VCF header for this builder.  This should be set before adding variants. */
+  def setHeader(header: VCFHeader): this.type = {
+    this._header = header
+    this
+  }
+
+  /** Sets the sequence dictionary for this builder.  This should be set before adding variants. */
+  def setSequenceDictionary(dict: SAMSequenceDictionary): this.type = {
+    this._header.setSequenceDictionary(dict)
+    this
+  }
+
+  /** Adds the header line to the header for this builder.  This should be set before adding variants. */
+  def addMetaDataLine(headerLine: VCFHeaderLine): this.type = {
+    this._header.addMetaDataLine(headerLine)
+    this
+  }
+
+  /** Add a variant. */
+  def addVariant(refIdx: Int = 0,
+                 start: Long,
+                 variantAlleles: List[String],
+                 genotypeAlleles: List[String] = List.empty,
+                 sampleName: Option[String] = None,
+                 phased: Boolean = false): this.type = {
+    val contig          = this.dict.getSequence(refIdx).getSequenceName
+    val alleles         = toAlleles(variantAlleles)
+    val stop            = VariantContextUtils.computeEndFromAlleles(alleles, start.toInt, -1)
+    val (ctxBuilder, prevGenotypes) = this.variants.find { ctx =>
+      ctx.getContig == contig &&
+        ctx.getStart == start &&
+        ctx.getEnd == stop &&
```

Comment:
Actually, [VCF 4.2](https://samtools.github.io/hts-specs/VCFv4.2.pdf) says "_It is permitted to have multiple records with the
same POS._" in section 1.4.1